presto-main/src/main/java/com/facebook/presto/operator/scalar/MapGenericEquality.java

```diff
@@ -34,36 +31,28 @@ Boolean equals(int leftMapIndex, int rightMapIndex)
     }
 
     public static Boolean genericEqual(
-            MethodHandle keyEqualsFunction,
-            MethodHandle keyHashcodeFunction,
             Type keyType,
             Block leftMapBlock,
             Block rightMapBlock,
             EqualityPredicate predicate)
     {
-        Map<KeyWrapper, Integer> wrappedLeftMap = new LinkedHashMap<>();
-        for (int position = 0; position < leftMapBlock.getPositionCount(); position += 2) {
-            wrappedLeftMap.put(new KeyWrapper(readNativeValue(keyType, leftMapBlock, position), keyEqualsFunction, keyHashcodeFunction), position + 1);
-        }
-
-        Map<KeyWrapper, Integer> wrappedRightMap = new LinkedHashMap<>();
-        for (int position = 0; position < rightMapBlock.getPositionCount(); position += 2) {
-            wrappedRightMap.put(new KeyWrapper(readNativeValue(keyType, rightMapBlock, position), keyEqualsFunction, keyHashcodeFunction), position + 1);
-        }
-
-        if (wrappedLeftMap.size() != wrappedRightMap.size()) {
+        if (leftMapBlock.getPositionCount() != rightMapBlock.getPositionCount()) {
             return false;
         }
 
-        for (Map.Entry<KeyWrapper, Integer> entry : wrappedRightMap.entrySet()) {
-            KeyWrapper key = entry.getKey();
-            Integer leftValuePosition = wrappedLeftMap.get(key);
-            if (leftValuePosition == null) {
+        SingleMapBlock mapLeftBlock = (SingleMapBlock) leftMapBlock;
```

Comment:
nit: I would call it `leftSingleMapBlock`, and make the argument parameter name to be `leftBlock`. ditto for rightBlock